javascripts/govuk/shim-links-with-button-role.js

```diff
@@ -16,40 +14,16 @@
 
   GOVUK.shimLinksWithButtonRole = {
 
-    // default configuration that can be overridden by passing object as second parameter to module
-    config: {
-      // the target element(s) to attach the shim event to
-      selector: 'a[role="button"]',
-      // array of keys to match against upon the keyup event
-      keycodes: [
-        32 // spacekey
-      ]
-    },
-
-    // event behaviour (not a typical anonymous function for resuse if needed)
-    triggerClickOnTarget: function triggerClickOnTarget (event) {
-      // if the code from this event is in the keycodes array then
-      if ($.inArray(event.which, this.config.keycodes) !== -1) {
-        event.preventDefault()
-        // trigger the target's click event
-        event.target.click()
-      }
-    },
-
-    // By default this will find all links with role attribute set to
-    // 'button' and will trigger their click event when the space key (32) is pressed.
-    // @method init
-    // @param  {Object}   customConfig                object to override default configuration
-    //         {String}   customConfig.selector       a selector for the elements to be 'clicked'
-    //         {Array}    customConfig.keycodes       an array of javascript keycode values to match against that when pressed will trigger the click
-    init: function init (customConfig) {
-      // extend the default config with any custom attributes passed in
-      this.config = $.extend(this.config, customConfig)
-      // if we have found elements then:
-      if ($(this.config.selector).length > 0) {
-        // listen to 'document' for keyup event on the elements and fire the triggerClickOnTarget
-        $(document).on('keyup', this.config.selector, this.triggerClickOnTarget.bind(this))
-      }
+    init: function init () {
+      // listen to 'document' for keydown event on the any elements that should be buttons.
+      $(document).on('keydown', '[role="button"]', function (event) {
+        // if the keyCode (which) is 32 it's a space, let's simulate a click.
+        if (event.which === 32) {
```

Comment:
Rather than the comment I’d have written something like: `var SPACEBAR = 32;` and `if (event.which === SPACEBAR) {` etc.